src/main/python/keycodes.py

```diff
@@ -556,6 +539,11 @@ def generate_keycodes_for_mask(label, mask):
         lbl = "M{}".format(x)
         KEYCODES_MACRO.append(Keycode(0x5F12 + x, lbl, lbl))
 
+    KEYCODES_USER.clear()
+    for x in range(len(keyboard.custom_keycodes)):
+        c_keycode = keyboard.custom_keycodes[x]
+        KEYCODES_USER.append(Keycode(0x5F80 + x, c_keycode.shortName, c_keycode.name, c_keycode.title))
```

Comment:
here when reading out of the dict, make sure the property actually exists. For example, using `key.get("shortName", "USER{:02}".format(x))`